lib/runtime/src/ops/layer_norm.cc

```diff
@@ -16,566 +16,197 @@
-  ln->add_int_vector_property("axes", axes);
-  ln->add_float_property("eps", eps);
-  layers.push_back(ln);
-  return ln->outputs[0];
+  return {LAYERNORM_INIT_TASK_ID, b};
 }
 
-Op *LayerNorm::create_operator_from_layer(
-    FFModel &model,
-    Layer const *layer,
-    std::vector<ParallelTensor> const &inputs) {
-  long long value;
-  layer->get_int_property("elementwise_affine", value);
-  bool elementwise_affine = (bool)value;
-  std::vector<int> axes;
-  layer->get_int_vector_property("axes", axes);
-  float eps;
-  layer->get_float_property("eps", eps);
-  return new LayerNorm(model,
-                       layer->layer_guid,
-                       inputs[0],
-                       axes,
-                       elementwise_affine,
-                       eps,
-                       false, // allocate_weights
-                       layer->name);
+OpTaskInvocation forward(LayerNormAttrs const &attrs) {
+  OpTaskBinding b;
+
+  b.bind(INPUT, input_tensor(0));
+  b.bind(OUTPUT, output_tensor(0));
+  b.bind(GAMMA, weight_tensor(0)); // todo, this may have some problem
+  b.bind(BETA, weight_tensor(1));  // how to get gmmam and beta
+  b.bind_arg(PROFILING, profiling_settings());
+  b.bind_arg(PER_DEVICE_STATE, per_device_state<LayerNormPerDeviceState>());
+
+  return {LAYERNORM_FWD_TASK_ID, b};
 }
 
-LayerNorm::LayerNorm(FFModel &model,
-                     LayerNormParams const &params,
-                     ParallelTensor const input,
-                     char const *name,
-                     bool allocate_weights)
-    : LayerNorm(model,
-                params.layer_guid,
-                input,
-                params.axes,
-                params.elementwise_affine,
-                params.eps,
-                allocate_weights,
-                name) {}
-
-LayerNorm::LayerNorm(FFModel &model,
-                     LayerID const &_layer_guid,
-                     const ParallelTensor _input,
-                     std::vector<int> const &_axes,
-                     bool _elementwise_affine,
-                     float _eps,
-                     bool allocate_weights,
-                     char const *name)
-    : Op(model,
-         OP_LAYERNORM,
-         _input->data_type,
-         name,
-         1 /*inputs*/,
-         _elementwise_affine ? 2 : 0 /*weights*/,
-         1 /*outputs*/,
-         _input),
-      elementwise_affine(_elementwise_affine), eps(_eps), axes(_axes) {
-  // overwrite layer_guid
-  layer_guid = _layer_guid;
-  outputs[0] = model.create_parallel_tensor_legion_ordering(
-      _input->num_dims, _input->dims, _input->data_type, this);
-  assert(check_output_input_weight_parallel_dims(allocate_weights));
-  ParallelDim output_dims[MAX_TENSOR_DIM];
-  int M = 1;
-  for (int i = 0; i < axes.size(); i++) {
-    M *= inputs[0]->dims[inputs[0]->num_dims - 1 - axes[i]].size;
-  }
-  effective_num_elements = M;
-  effective_batch_size = inputs[0]->get_volume() / M;
```

Comment:
@lambda7xx this is the original code that sets `effective_num_elements` and `effective_batch_size`. Seems like we can get this from `input`